src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -159,10 +159,28 @@ public override void CancelPendingRead()
 
         /// <inheritdoc />
         public override void Complete(Exception? exception = null)
+        {
+            if (CompleteAndGetNeedsDispose())
+            {
+                InnerStream.Dispose();
+            }
+        }
+
+#if NETCOREAPP
```

Comment:
Why is this only on NETCOREAPP?

Reply:
Because `Stream.DisposeAsync()` isn't available on net462 or netstandard2.0